### PR TITLE
Issue id#27 qa improvements

### PR DESCRIPTION
**Pull request by snlacks**
Changes to note: 

1. Changed the prod-module-boilerplate for current build, with accomodations for grunt, no ES6/Babel, etc. 
2. Dependency-check module included as dependency, but could not get it the script to run. Error messages were not helpful for me.
3. gruntfile has options set for the with statement in the constantTimeCompare.js, with-statement isn't allowed with "use strict", but options for rest of app "use strict" so I included the second jshint task for that file by itself but with different options.
4. Copied the webpack config over from prod-module-boilerplate, changed the entry file.

### CODE REVIEW

File: gruntfile.js [hunk 1]
```diff
@@ -4,23 +4,45 @@ module.exports = function(grunt) {
 
   grunt.initConfig({
```
Comment (ericelliott) [inline gruntfile.js]: We have ESLint, we don't need grunt to lint anymore.

File: .jshintrc [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (ericelliott) [inline .jshintrc]: We should be able to delete this now.

**Comment by ericelliott** [comment]
> with accomodations for grunt

We don't need Grunt anymore with ESLint. Speaking of which, I don't see the `prod-module-boilerplate` `.eslintrc`.

> with-statement isn't allowed with "use strict"

We'll need to remove `with` from constantEquals. It doesn't exist in ES6 (upgrading to ES6 is the next modernization step).

**Comment by snlacks** [comment]
I've updated these, and made sure all tests were passing. I turned off the no-shadow option because it was showing an error for:

function(stuff, function(err){
   function(other_stuff, function(err){
   //
   });
});
 
Oddly enough, constantTimeCompare.js is passing, it shouldn't be.

Also, in package.json, the new pattern made multiple pre-commits redundant, so I just have it run validate, which runs test and lint, removing the superfluous test and lint.

**Comment by ericelliott** [comment]
> Oddly enough, constantTimeCompare.js is passing, it shouldn't be.

Weird.

**Comment by ericelliott** [comment]
Please add [`.travis.yml`](https://github.com/ericelliott/prod-module-boilerplate/blob/master/.travis.yml)

**Comment by snlacks** [comment]
> Oddly enough, constantTimeCompare.js is passing, it shouldn't be.

>> Weird.

The with statement isn't causing an error. I played with it a bunch more, got no where, and opened an [issue](https://github.com/eslint/eslint/issues/2925) on eslint

> Please add [.travis.yml](https://github.com/ericelliott/prod-module-boilerplate/blob/master/.travis.yml)

[done](https://github.com/snlacks/credential/blob/IssueID%2327-QA-Improvements/.travis.yml)

Let me know if I missed anything else, but I triple checked this time. Thanks for your patience.

**Comment by ericelliott** [comment]
Please push one more commit to trigger Travis.
It can just be a commit message like "Trigger Travis build".

**Comment by ericelliott** [comment]
Nevermind. Merged. If it's broke, we'll fix it. =)

**Comment by snlacks** [comment]
Already triggered.

**Pull request merged**